modules/basegl/src/processors/imageprocessing/imagesubsetgl.cpp

```diff
@@ -0,0 +1,239 @@
+
+    inport_.onChange([this]() {
+        NetworkLock lock(this);
+
+        // Update to the new dimensions.
+        dims_ = inport_.getData()->getDimensions();
+
+        rangeX_.setRangeNormalized(size2_t(0, dims_.x));
+        rangeY_.setRangeNormalized(size2_t(0, dims_.y));
+
+        // set the new dimensions to default if we were to press reset
+        rangeX_.setCurrentStateAsDefault();
+        rangeY_.setCurrentStateAsDefault();
+    });
+
+    // Create a rectangle mesh for rendering.
+    // Texture coordinates will be modified
+    auto verticesBuffer =
+        util::makeBuffer<vec2>({{-1.0f, -1.0f}, {1.0f, -1.0f}, {-1.0f, 1.0f}, {1.0f, 1.0f}});
+    vec2 texOffset = vec2(0.f) / vec2(dims_);
+    vec2 texEnd = vec2(1.f) / vec2(dims_);
+    texCoordsBuffer_ = util::makeBuffer<vec2>(
+        {texOffset, {texEnd.x, texOffset.y}, {texOffset.x, texEnd.y}, texEnd});
+    auto indices = util::makeIndexBuffer({0, 1, 2, 3});
+    rect_.addBuffer(BufferType::PositionAttrib, verticesBuffer);
+    rect_.addBuffer(BufferType::TexcoordAttrib, texCoordsBuffer_);
+    rect_.addIndicies(Mesh::MeshInfo(DrawType::Triangles, ConnectivityType::Strip), indices);
+}
+
+ImageSubsetGL::~ImageSubsetGL() = default;
+
+void ImageSubsetGL::process() {
+    const ivec2 offset{rangeX_.get().x, rangeY_.get().x};
+    const ivec2 dim = ivec2{rangeX_.get().y, rangeY_.get().y} - offset;
+    if (dim == ivec2(dims_))
+        outport_.setData(inport_.getData());
+    else {
+        auto image = inport_.getData();
+        auto outImage = std::make_shared<Image>(dim, image->getDataFormat());
+        // pass meta data on
+        outImage->copyMetaDataFrom(*image);
+
+        outport_.setData(outImage);
+
+        TextureUnit::setZeroUnit();
+
+        TextureUnit colorUnit, depthUnit, pickingUnit;
+        utilgl::activateAndClearTarget(outport_, ImageType::ColorDepthPicking);
+
+        shader_.activate();
+        shader_.setUniform("color_", colorUnit.getUnitNumber());
+        shader_.setUniform("depth_", depthUnit.getUnitNumber());
+        shader_.setUniform("picking_", pickingUnit.getUnitNumber());
+
+        utilgl::bindTextures(*image, colorUnit, depthUnit, pickingUnit);
+        // Texture coordinates of the region we should copy
+        vec2 texOffset = vec2(offset) / vec2(dims_);
+        vec2 texEnd = vec2(offset + dim) / vec2(dims_);
+        auto coords = texCoordsBuffer_->getEditableRAMRepresentation();
+        coords->set(0, texOffset);
+        coords->set(1, {texEnd.x, texOffset.y});
+        coords->set(2, {texOffset.x, texEnd.y});
+        coords->set(3, texEnd);
+
+        utilgl::Enable<MeshGL> enable(rect_.getRepresentation<MeshGL>());
+        utilgl::DepthFuncState depth(GL_ALWAYS);
+        glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
+
+        shader_.deactivate();
+        utilgl::deactivateCurrentTarget();
+        TextureUnit::setZeroUnit();
+    }
+}
+
+void ImageSubsetGL::invokeEvent(Event* event) {
```

Comment:
We have event properties for this. It is much better to use them than to override the invokeEvent method.

Reply:
See general comment